config/routes.rb

```diff
@@ -1,10 +1,18 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  devise_for :users
+  devise_scope :user do
+    get '/users/sign_out' => 'devise/sessions#destroy'
+  end
+   
+    resources :foods do
+      resources :recipe_foods
+    end
+    resources :public_recipes
+    resources :shopping_list
```

Comment:
- [ ] Kindly ensure that this routes name is according to what we have in the general_shopping_list controller. 👍 I was unable to access your app because of this error.

```suggestion
    resources :general_shopping_lists
```